pkg/resource/deploy/providers/reference.go

```diff
@@ -41,6 +41,11 @@ func IsProviderType(typ tokens.Type) bool {
 	return typ.Module() == "pulumi:providers" && typ.Name() != ""
 }
 
+// IsDefaultProvider returns true if this URN refers to a default Pulumi provider.
+func IsDefaultProvider(urn resource.URN) bool {
+	return IsProviderType(urn.Type()) && strings.HasPrefix(urn.Name().String(), "default")
```

Comment:
Do we have code somewhere to ban `default` as a prefix for user-instantiated providers?

Reply:
I don't believe we do, for `default` or otherwise.

Reply:
I also do not believe we have such things.  I think it would be reasonable for us to ban creating of first class providers which have names of this shape, but I would also be fine doing that later.